src/main/java/uk/gov/pay/api/agreement/resource/AgreementsApiResource.java

```diff
@@ -57,17 +53,20 @@ public class AgreementsApiResource {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AgreementsApiResource.class);
 
-    private final AgreementService agreementService;
-    private final LedgerService ledgerService;
+    private final CreateAgreementService createAgreementService;
```

Comment:
Any reason we need a different agreement service for each action? Can we not have one AgreementService with method for each action?

Reply:
We can, ofc. We were thinking that have everything like this, split into actions, eg `CreatePaymentService`, `CancelPaymentService`, `GetPaymentRefundService`, etc.

Reply:
Payments/refunds are a bit complex because of URI building and have additional logic to go to the connector or ledger so separate services classes are fine.

not sure if agreements need such granular services